src/parse/errors.rs

```diff
@@ -116,13 +116,30 @@ pub enum ErrorKind {
     /// let res = App::new("prog")
     ///     .arg(Arg::new("color")
     ///          .setting(ArgSettings::TakesValue)
+    ///          .setting(ArgSettings::NoEmptyValues)
     ///          .long("color"))
     ///     .try_get_matches_from(vec!["prog", "--color="]);
     /// assert!(res.is_err());
     /// assert_eq!(res.unwrap_err().kind, ErrorKind::EmptyValue);
     /// ```
     EmptyValue,
 
+    /// Occurs when the user doesn't use equals for an option that requres equal
+    /// sign to provide values.
+    ///
+    /// ```rust
+    /// # use clap::{App, Arg, ErrorKind, ArgSettings};
+    /// let res = App::new("prog")
+    ///     .arg(Arg::new("color")
+    ///          .setting(ArgSettings::TakesValue)
+    ///          .setting(ArgSettings::RequireEquals)
+    ///          .long("color"))
+    ///     .try_get_matches_from(vec!["prog", "--color", "red"]);
+    /// assert!(res.is_err());
+    /// assert_eq!(res.unwrap_err().kind, ErrorKind::NoEquals);
+    /// ```
+    NoEquals,
```

Comment:
What happens when the arg has a default?

Reply:
Using an option which has a default value without providing value, an `EmptyValue` will emits. If users sets `default_missing_value`, using option without providing value is acceptable.

Reply:
What? Maybe we are misunderstanding each other here. Are you saying that `--foo` will set the value as `''` when `RequiredEquals` is set?

Reply:
Nope, that not my opinion. For `Arg::new("foo").long("foo").set(RequiredEquals)`, using `test-app --foo` will emits `NoEquals` error. Check the test `no_empty_values_without_equals `.

Reply:
But the arg doesn't have `takes_value`, right? Shouldn't that be successful?

And what happens if it has `default_value(2)` after setting `takes_value`?

Reply:
`RequiredEquals` documentation: Setting this implies [`ArgSettings::TakesValue`].

Reply:
Sorry, yeah I forgot. But yeah, what happens if I do `default_value` now? This was my original question.

Reply:
`default_value` only works when argument is not provided by the end user. So setting `default_value` doesn't change anything here. Do you mean `default_missing_value`?

Reply:
With `default_missing_value` set, there will be no error.

Reply:
Yup, that is what I wanted to confirm. Thanks 😄